log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala

```diff
@@ -118,4 +120,28 @@ class SolrCloudStreamTest extends FunSuite with BaseSolrCloudTest {
     assert(streamFailed)
   }
 
+  test("should flush after n events") {
+    val stream = new TestStream(1, 10)
+
+    (1 to 10).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 10)
+  }
+
+  test("should flush after n seconds") {
+    val stream = new TestStream(1, 10)
+
+    (1 to 3).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 3)
+  }
+}
+
+class TestStream(flushDuration: Int, flushSize: Int) extends Stream[String](StreamParams()) {
```

Comment:
Let's put the `Duration` into the argument. You can do `import scala.concurrent.duration._`, then create a duration like `1 second`. This also disambiguates the two arguments, it's easy to make errors when you have two int arguments next to eachother, it's also harder to read

Reply:
StreamParams takes flushDuration as Int. So the argument is kept  as Int. Direct conversion from Duration to Int is not there. Can I keep it like this or I try ways for this conversion

Reply:
let's leave it the way it is for now